streams/src/main/java/org/apache/kafka/streams/processor/api/ProcessorContext.java

```diff
@@ -247,4 +251,45 @@ Cancellable schedule(final Duration interval,
      * @return the key/values matching the given prefix from the StreamsConfig properties.
      */
     Map<String, Object> appConfigsWithPrefix(final String prefix);
+
+    /**
+     * Return the current system timestamp (also called wall-clock time) in milliseconds.
+     *
+     * <p> Note: this method returns the internally cached system timestamp from the Kafka Stream runtime.
+     * Thus, it may return a different value compared to {@code System.currentTimeMillis()}.
+     *
+     * @return the current system timestamp in milliseconds
+     */
+    long currentSystemTimeMs();
```

Comment:
Did we miss to add those to the new context via https://cwiki.apache.org/confluence/display/KAFKA/KIP-622%3A+Add+currentSystemTimeMs+and+currentStreamTimeMs+to+ProcessorContext ???

Reply:
Yeah, I think it was just a miss. That KIP might have been going in parallel with the new PAPI, I don't remember.

I agree adding these would just be an update to KIP-622 instead of a part of this KIP. I'll just go ahead and send a separate PR for this and a note to the KIP-622 vote thread.

Reply:
Fixed via https://issues.apache.org/jira/browse/KAFKA-13699